src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/CollectionsShouldImplementGenericInterfaceTests.cs

```diff
@@ -12,12 +12,366 @@ public class CollectionsShouldImplementGenericInterfaceTests : DiagnosticAnalyze
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicCollectionsShouldImplementGenericInterfaceAnalyzer();
+            return new CollectionsShouldImplementGenericInterfaceAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpCollectionsShouldImplementGenericInterfaceAnalyzer();
+            return new CollectionsShouldImplementGenericInterfaceAnalyzer();
+        }
+
+        [Fact]
+        public void Test_WithCollectionBase()
+        {
+            #region CSharp Test
+            VerifyCSharp(@"
+                        using System;
+                        using System.Collections;
+                        using System.Collections.Generic;
+                        public class TestClass :CollectionBase
+                        {
+                            public int Count
+                            {
+                                get
+                                {
+                                    throw new NotImplementedException();
+                                }
+                            }
+                        }", 
+                        GetCSharpResultAt(5, 38, CollectionsShouldImplementGenericInterfaceAnalyzer.RuleId, CollectionsShouldImplementGenericInterfaceAnalyzer.Rule.MessageFormat.ToString()));
```

Comment:
Since this is repeated so many times, and is so long, I'd put this at the top of the class:
```
private static readonly ErrorMessage = CollectionsShouldImplementGenericInterfaceAnalyzer.RuleId, CollectionsShouldImplementGenericInterfaceAnalyzer.Rule.MessageFormat.ToString();
```
... and use that everywhere.